modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,164 @@
+// Module included in the following assemblies:
+//
+// * oadp-api
+
+:_content-type: REFERENCE
+[id="oadp-api-details_{context}"]
+= OADP API
+
+The following tables expose the structure of the OADP API:
+
+.DataProtectionApplicationSpec
+[options="header"]
+|===
+|Property|Type|Description
+
+|`BackupLocations`
+|[] `BackupLocation`
+|Defines the list of desired configurations to use for `BackupStorageLocations`.
+
+|`SnapshotLocations`
+|[] `SnapshotLocation`
```

Comment:
maybe add a link [][SnapshotLocation](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#SnapshotLocation) or hyperlink to `Table 3. SnapshotLocation`

Reply:
Done.